enterprise/server/util/cacheproxy/cacheproxy_test.go

```diff
@@ -179,6 +180,7 @@ func TestWriter(t *testing.T) {
 		if err != nil {
 			t.Fatal(err)
 		}
+		defer r.Close()
```

Comment:
Because this is inside a for loop, the defer won't be closed until after all loops are executed and the test finishes. 

 don't know if that will cause a problem here or not, but it might be better to explicitly call r.Close() after we''ve used it in digest.Compute() -- so like:

        ...
		d2, err := digest.Compute(r)
		if err != nil {
			t.Fatal(err)
		}
        r.Close()
        ...

(here, and below 2 instances as well)

Reply:
So instead of moving the call and removing the defer, which I think leads to more difficult to maintain code, I wrapped the section that uses the Reader in a closure to properly scope the defer. Lemme know if there's something wrong with this approach!